src/components/editor/property/InputLookupQA.jsx

```diff
@@ -23,16 +23,39 @@ const AsyncTypeahead = asyncContainer(Typeahead)
 //  based on values in propertyTemplate.valueConstraint.useValuesFrom
 //  and the lookupConfig for the URIs has component value of 'lookup'
 const InputLookupQA = (props) => {
-  const [authorities, setAuthorities] = useState([])
+  const triggerRenderState = useState('')
+  const setTriggerRender = triggerRenderState[1]
+  // Using a ref so that can append to current list of authorities.
+  const allAuthorities = useRef([])
+  // Tokens allow us to cancel an existing search. Does not actually stop the
+  // search, but causes result to be ignored.
+  const tokens = useRef([])
 
   const search = (query) => {
-    const currentAuthorities = []
-    const resultPromise = getSearchResults(query, props.propertyTemplate)
-    resultPromise.then((result) => {
-      result.forEach((authority) => {
-        currentAuthorities.push(authority)
+    // Clear the authorities.
+    // No re-render, so change not visible to user.
+    allAuthorities.current = []
+
+    // Cancel all current searches
+    while (tokens.current.length > 0) {
+      tokens.current.pop().cancel = true
+    }
+
+    // Create a token for this set of searches
+    const token = { cancel: false }
+    tokens.current.push(token)
+
+    // resultPromises is an array of Promise<authority>
+    const resultPromises = getSearchResults(query, props.propertyTemplate)
+    resultPromises.forEach((resultPromise) => {
+      resultPromise.then((authority) => {
+        // Only use these results if not cancelled.
+        if (!token.cancel) {
+          allAuthorities.current.push(authority)
+          // Changing state triggers re-render.
+          setTriggerRender(authority)
```

Comment:
Would there ever be a problem here if the `authority` for the last search is the same as the first authority for the next search? Particularly when a search only uses one authority?